### PR TITLE
Add IsNullOrWhitespaceCondition

### LINKED ISSUE #124: Add IsNullOrWhitespace Condition
I just needed an `IsNullOrWhitespace` condition, because I had to wait until an HTML attribute was available.

Would you be interested in adding it to boa itself? I can submit a PR with it later.

### PR DESCRIPTION
**Pull request by SabotageAndi**
resolves #124 

@AndyLPK247 I tried to follow everything from the contribution guidelines. Please let me know if I missed something.

### CODE REVIEW

File: Boa.Constrictor/Screenplay/Conditions/Strings/IsNullOrWhitespaceCondition.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
+﻿namespace Boa.Constrictor.Screenplay
+{
+    public class IsNullOrWhitespaceCondition : ICondition<string>
```
Comment (pl-aknight) [inline Boa.Constrictor/Screenplay/Conditions/Strings/IsNullOrWhitespaceCondition.cs:3]: For readability, please change this class name to `IsNullOrWhitespace`. That mirrors the naming of other Conditions.

Example: `Actor.WaitsUntil(Text.Of(something), IsNullOrWhitespace.Value())`

File: Boa.Constrictor/Screenplay/Conditions/Strings/IsNullOrWhitespaceCondition.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        #region Methods
+
+        /// <summary>
+        /// Checks if the actual value equals null or whitespace
+        /// </summary>
+        /// <param name="actual"></param>
+        /// <returns></returns>
+        public bool Evaluate(string actual)
+        {
+            return string.IsNullOrWhiteSpace(actual);
```
Comment (pl-aknight) [inline Boa.Constrictor/Screenplay/Conditions/Strings/IsNullOrWhitespaceCondition.cs:37]: For consistent coding style, please use the `=>` operator for defining this method.

File: Boa.Constrictor.UnitTests/Screenplay/Conditions/Strings/IsNullOrWhitespaceConditionTests.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
+﻿using Boa.Constrictor.Screenplay;
+using FluentAssertions;
+using NUnit.Framework;
+
+namespace Boa.Constrictor.UnitTests.Screenplay
+{
+    [TestFixture]
+    public class IsNullOrWhitespaceConditionTests
```
Comment (pl-aknight) [inline Boa.Constrictor.UnitTests/Screenplay/Conditions/Strings/IsNullOrWhitespaceConditionTests.cs:8]: Thanks for writing unit tests! This is awesome.

Please update the class name to align with the new Condition name: `IsNullOrWhitespaceTests`.

**Review by pl-aknight — CHANGES_REQUESTED** [review]
Great work, @SabotageAndi ! Thanks so much for contributing a new Condition. My one piece of feedback is naming. Please update, and I'll approve. I'll also release a new NuGet package thereafter.

File: Boa.Constrictor/Screenplay/Conditions/Strings/IsNullOrWhitespaceCondition.cs (hunk 1 above)
Comment (SabotageAndi) [reply Boa.Constrictor/Screenplay/Conditions/Strings/IsNullOrWhitespaceCondition.cs]: Oops, it is renamed now.

File: Boa.Constrictor/Screenplay/Conditions/Strings/IsNullOrWhitespaceCondition.cs (hunk 2 above)
Comment (SabotageAndi) [reply Boa.Constrictor/Screenplay/Conditions/Strings/IsNullOrWhitespaceCondition.cs]: Done

File: Boa.Constrictor.UnitTests/Screenplay/Conditions/Strings/IsNullOrWhitespaceConditionTests.cs (hunk 3 above)
Comment (SabotageAndi) [reply Boa.Constrictor.UnitTests/Screenplay/Conditions/Strings/IsNullOrWhitespaceConditionTests.cs]: Also changed this

**Review by pl-aknight — APPROVED** [review]
Great work, @SabotageAndi!

**Pull request merged**